odl/discr/diff_ops.py

```diff
@@ -802,7 +802,12 @@ def finite_diff(f, axis, dx=1.0, method='forward', out=None, **kwargs):
     if pad_mode not in _SUPPORTED_PAD_MODES:
         raise ValueError('`pad_mode` {} not understood'
                          ''.format(pad_mode))
-    pad_const = float(kwargs.pop('pad_const', 0))
+
+    pad_const = kwargs.pop('pad_const', 0)
+    if np.iscomplexobj(f):
```

Comment:
This should be written `pad_const = f.dtype.type(pad_const)` in order to get the correct datatype (e.g. `float32` vs `float64`).

Reply:
Sure, that is a better approach.